functions/Export-DbaDacpac.ps1

```diff
@@ -0,0 +1,179 @@
+	https://msdn.microsoft.com/en-us/library/hh550080.aspx
+	
+
+	.PARAMETER EnableException
+	By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+	This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+	Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+	.NOTES
+	Tags: migration, database 
+	Author: Richie lee (@bzzzt_io)
+
+	Website: https://dbatools.io
+	Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+	License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+	
+	.LINK
+	https://dbatools.io/Export-DbaDacpac
+
+	.EXAMPLE
+	Export-DbaDacpac -SqlInstance sql2016 -Database SharePoint_Config
+	Exports the dacpac for SharePoint_Config on sql2016 to $home\Documents\SharePoint_Config.dacpac
+
+	.EXAMPLE
+	$moreprops = "/p:VerifyExtraction=$true /p:CommandTimeOut=10"
+	Export-DbaDacpac -SqlInstance sql2016 -Database SharePoint_Config -Directory C:\temp -ExtendedProperties $moreprops
+
+	Sets the CommandTimeout to 10 then extracts the dacpac for SharePoint_Config on sql2016 to C:\temp\SharePoint_Config.dacpac then verifies extraction.
+
+	
+    #>
+	[CmdletBinding()]
+	param
+	(
+		[parameter(Mandatory, ValueFromPipeline)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[Alias("Credential")]
+		[PSCredential]$SqlCredential,
+		[object[]]$Database,
+		[object[]]$ExcludeDatabase,
+		[switch]$AllUserDatabases,
+		[string]$Directory = "$home\Documents",
+		[string]$ExtendedParameters,
+		[string]$ExtendedProperties,
+		[switch]$EnableException
+	)
+	
+	process {
+		if (!$Database -and !$ExcludeDatabase -and !$AllUserDatabases) {
+			Stop-Function -Message "You must specify databases to execute against using either -Database, -ExcludeDatabase or -AllUserDatabases" -Continue
+		}
+		
+		if (-not (Test-Path $Directory)) {
+			Stop-Function -Message "$Directory doesn't exist or access denied" -Continue
+		}
+		
+		foreach ($instance in $sqlinstance) {
+			
+			try {
+				Write-Message -Level Verbose -Message "Connecting to $instance."
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential -MinimumVersion 9
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+			
+			$dbs = $server.Databases | Where-Object { $_.IsSystemObject -eq $false }
+			
+			if ($Database) {
+				$dbs = $dbs | Where-Object Name -in $Database
+			}
+			
+			if ($ExcludeDatabase) {
+				$dbs = $dbs | Where-Object Name -notin $ExcludeDatabase
+			}
+			
+			foreach ($db in $dbs) {
+				$dbname = $db.name
+				$connstring = $server.ConnectionContext.ConnectionString.Replace('"', "'")
+				if ($connstring -notmatch 'Database=') {
+					$connstring = "$connstring;Database=$dbname"
+				}
+				$path = "$Directory\$dbname.dacpac"
```

Comment:
Since we can suply multiple instances, there's a chance we'll have DB's with the same name.
Won't it be safer to include the '\computername\instancename' in the path?

Reply:
good call! ill do instance with \ replaced with $